src/directions.js

```diff
@@ -15,6 +17,36 @@ var Directions = L.Class.extend({
         this._waypoints = [];
     },
 
+    geocode: function(waypoint, proximity, cb) {
+        this._geocode = corslite(L.Util.template(Directions.GEOCODER_TEMPLATE, {
+            query: waypoint.properties.query,
+            token: this.options.accessToken || L.mapbox.accessToken,
+            proximity: proximity && [proximity.lng, proximity.lat].join(',')
+        }), L.bind(function (err, resp) {
+            this._geocode = null;
+
+            if (err && err.type === 'abort') {
+                return cb(err);
+            }
+
+            resp = resp || err;
+
+            if (resp && resp.responseText) {
+                try {
+                    resp = JSON.parse(resp.responseText);
+                } catch (e) {
+                    return cb(e);
+                }
+            }
+
+            if (resp && resp.features && resp.features.length > 0) {
+                waypoint.geometry.coordinates = resp.features[0].center;
+                waypoint.properties.name = resp.features[0].place_name;
+                return cb(null);
+            }
```

Comment:
The `else` case here likely needs to be turned into a "No results found for 'query'" error.